### PR TITLE
Removed dead code from System.Data.Common

### PR DESCRIPTION
**Pull request by YoupHulsebos**
Removed dead code for System.Data.Common for issue #17905.

### CODE REVIEW

File: src/System.Data.Common/src/System.Data.Common.csproj [hunk 1]
```diff
@@ -56,7 +55,9 @@
     <Compile Include="System\Data\ConstraintCollection.cs" />
     <Compile Include="System\Data\ConstraintConverter.cs" />
     <Compile Include="System\Data\ConstraintEnumerator.cs" />
-    <Compile Include="System\Data\DataColumn.cs" />
+    <Compile Include="System\Data\DataColumn.cs">
+      <SubType>Component</SubType>
+    </Compile>
```
Comment (stephentoub) [inline src/System.Data.Common/src/System.Data.Common.csproj]: Looks like VS may have made some edits here?  Can you remove the ones like this that weren't intentional?

Comment (YoupHulsebos) [inline src/System.Data.Common/src/System.Data.Common.csproj]: Sure thing. Fixed them in the new commit

Comment (YoupHulsebos) [inline src/System.Data.Common/src/System.Data.Common.csproj]: Thought I got it. But VS got me again.

Comment (YoupHulsebos) [inline src/System.Data.Common/src/System.Data.Common.csproj]: Finaly, its fixed :)

**Comment by karelz** [comment]
I updated the top post of #17905. Thanks @YoupHulsebos!
cc @danmosemsft

**Comment by karelz** [comment]
cc @saurabh500 @corivera

**Pull request merged**